packages/js/components/src/experimental-select-tree-control/select-tree.tsx

```diff
@@ -268,6 +297,46 @@ export const SelectTree = function SelectTree( {
 											props.onRemove( item );
 										}
 									} }
+									onBlur={ ( event ) => {
+										if ( isEventOutside( event ) ) {
+											setIsOpen( false );
+											setIsFocused( false );
+										}
+									} }
+									onKeyDown={ ( event ) => {
+										if (
+											event.key === 'ArrowLeft' ||
+											event.key === 'ArrowRight'
+										) {
```

Comment:
I don't think the tree control should have such in-depth knowledge of another components implementation. All of this selection of items in the select control should happen in the select control. The tree control should just pass the events via a callback prop passed into it, which can then pass things to the select control.